File: packages/cli/.generated/nori/nori.ts



type LanguageCode = (typeof LanguageCode)[keyof typeof LanguageCode];
const LanguageCode = Object.freeze({
	EnglishBritish: "en-GB",
	Japanese: "ja-JP",
});


export interface NoriI18nCollection extends Readonly<Record<LanguageCode, string>> {
	_meta: {
		kind: "noriI18nCollection";
	};
}

const createNoriI18nCollection = (
	collection: Readonly<Record<LanguageCode, string>>
): NoriI18nCollection => {
	return Object.freeze({
		...collection,
		_meta: {
			kind: "noriI18nCollection" as const
		}
	});
};

const createNoriI18nCollectionGenerator = <TParams extends Record<string, any>>(
	generator: (params: TParams) => NoriI18nCollection
): ((params: TParams) => NoriI18nCollection) => {
	return generator;
};


const nori = Object.freeze({


/** Collection: root */
root: {
/**
 * - en-GB: Displayed on first step of onboarding wizard.
 * - ja-JP: オンボーディングウィザードの最初のステップに表示されます。
 */letsGetStarted: createNoriI18nCollectionGenerator(({
topic,
}: {
/**
 * - en-GB: The topic to get started with.
 * - ja-JP: 始めるトピック。
 */
topic: string;
}) => createNoriI18nCollection({
[LanguageCode.EnglishBritish]: `Let's get started with ${ topic }.`,
[LanguageCode.Japanese]: `さあ、${ topic }を始めましょう！`
})),
},


/** Collection: client */
client: {
/**
 * - en-GB: A friendly greeting message.
 * - ja-JP: 親しみやすい挨拶メッセージ。
 */greeting: createNoriI18nCollection({
[LanguageCode.EnglishBritish]: `Hello!`,
[LanguageCode.Japanese]: `こんにちは！`
}),
/**
 * - en-GB: A friendly farewell message.
 * - ja-JP: 親しみやすい別れのメッセージ。
 */farewell: createNoriI18nCollection({
[LanguageCode.EnglishBritish]: `Goodbye!`,
[LanguageCode.Japanese]: `さようなら！`
})
},

});

export default nori;

File: packages/cli/.generated/nori/nori.py
from __future__ import annotations

from dataclasses import dataclass
from enum import StrEnum


class LanguageCode(StrEnum):
    ENGLISH_BRITISH = "en-GB"
    JAPANESE = "ja-JP"


@dataclass(frozen=True)
class NoriI18nCollection:
    english_british: str = ""
    japanese: str = ""

    def get(self, lang: LanguageCode) -> str:
        return getattr(self, lang.name.lower())


def _col(**kwargs: str) -> NoriI18nCollection:
    return NoriI18nCollection(**kwargs)

class root:
    """Collection: root"""

    @staticmethod
    def lets_get_started(*, topic: str) -> NoriI18nCollection:
      """
    en-GB: Displayed on first step of onboarding wizard.
    ja-JP: オンボーディングウィザードの最初のステップに表示されます。
      """
      return _col(
      english_british=f"Let's get started with {topic}.",
      japanese=f"さあ、{topic}を始めましょう！",
      )

class client:
    """Collection: client"""

    @staticmethod
    def greeting() -> NoriI18nCollection:
      """
    en-GB: A friendly greeting message.
    ja-JP: 親しみやすい挨拶メッセージ。
      """
      return _col(
      english_british="Hello!",
      japanese="こんにちは！",
      )

    @staticmethod
    def farewell() -> NoriI18nCollection:
      """
    en-GB: A friendly farewell message.
    ja-JP: 親しみやすい別れのメッセージ。
      """
      return _col(
      english_british="Goodbye!",
      japanese="さようなら！",
      )
